tests/test_tools.py

```diff
@@ -517,3 +518,12 @@ def test_latex_safe(input, expected, kwargs):
 def test_update_schemas(default_schema, user_schema, expected_schema):
     updated_schema = update_schemas(default_schema, user_schema)
     assert updated_schema == expected_schema
+
+
+def test_group():
+    doc0 = {"k0": "v00", "k1": "v01"}
```

Comment:
maybe test also for case where k0 and k1 is in a doc2

Reply:
Ok.